src/john.c

```diff
@@ -1813,6 +1823,11 @@ static void john_run(void)
 		if (trigger_reset)
 			database.format->methods.reset(&database);
 
+		if (!(options.flags & FLG_STDOUT) && john_main_process) {
+			john_log_format2();
+			log_flush();
+		}
```

Comment:
FWIW, I've confirmed that the `if` condition here is effectively the same as the one we use for calling `john_log_format()` above - it is.

I'm not happy about the `john_log_format` vs. `john_log_format2` function names, but I don't readily have a better suggestion.